scc/light_client/light_client.go

```diff
@@ -0,0 +1,56 @@
+package light_client
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/0xsoniclabs/consensus/inter/idx"
+	"github.com/0xsoniclabs/sonic/scc"
+	lc_state "github.com/0xsoniclabs/sonic/scc/light_client/light_client_state"
+	"github.com/0xsoniclabs/sonic/scc/light_client/provider"
+)
+
+// LightClient is the main entry point for the light client.
+// It is responsible for managing the light client state and
+// interacting with the provider.
+type LightClient struct {
+	provider provider.Provider
+	state    *lc_state.State
+}
+
+// Config is used to configure the LightClient.
+// It requires an url for the certificate providers and an initial committee.
+type Config struct {
+	Provider string
```

Comment:
nit: rename to "ProviderUrl" to indicate that this is expected to be a URL? Alternatively, use a URL type to indicate what this is supposed to be.

Reply:
https://github.com/0xsoniclabs/sonic/pull/120/commits/700da771b2480c34fb638f60ece472a6ed674899